ssl-config/src/test/java/com/palantir/ssl/SslSocketFactoriesConnectionTests.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.ssl;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLHandshakeException;
+import javax.net.ssl.SSLServerSocket;
+import javax.net.ssl.SSLServerSocketFactory;
+import javax.net.ssl.SSLSocketFactory;
+import jersey.repackaged.com.google.common.base.Throwables;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+/**
+ * Tests for {@link SslSocketFactories} that test that SSL connections
+ * can be established properly.
+ */
+public final class SslSocketFactoriesConnectionTests {
+
+    @Rule
+    public Timeout testTimeout = Timeout.seconds(5);
+
+    @Test
+    public void testSslNoClientAuthenticationJks() {
+        SslConfiguration serverConfig = SslConfiguration.builder()
+                .trustStorePath(TestConstants.CA_TRUST_STORE_PATH)
+                .keyStorePath(TestConstants.SERVER_KEY_STORE_JKS_PATH)
+                .keyStorePassword(TestConstants.SERVER_KEY_STORE_JKS_PASSWORD)
+                .build();
+
+        SslConfiguration clientConfig = SslConfiguration.builder()
+                .trustStorePath(TestConstants.CA_TRUST_STORE_PATH)
+                .build();
+
+        runSslConnectionTest(serverConfig, clientConfig, false);
```

Comment:
nit: instead of the boolean here, would be better to use an enum (would make the test clearer) or two methods `runSslConnectionTestWithoutClientAuth`, etc.